src/Components/GEOSldas_GridComp/GEOSmetforce_GridComp/LDAS_Forcing.F90

```diff
@@ -2011,6 +2022,55 @@ subroutine get_ERA5_LIS(date_time, met_path, N_catd, tile_coord, &
        i_ind(k) = ceiling( (this_lon - era5_grid_ll_lon)/era5_grid_dlon )
        j_ind(k) = ceiling( (this_lat - era5_grid_ll_lat)/era5_grid_dlat )
 
+      if (MET_HINTERP .eq. 1) then
+        ! pchakrab: For bilinear interpolation, for each tile, we need:
+        !  x1, x2, y1, y2 (defining the co-ords of four neighbors) and
+        !  i1, i2, j1, j2 (defining the indices of four neighbors)
+
+        ! the nearest neighbor forcing grid cell ("1") is already known.
+        ! It is the: (i_ind(k),j_ind(k))
+
+        ! wrap-around
+        if (i_ind(k)>era5_grid_N_lon) i_ind(k) = 1
+        if (j_ind(k)>era5_grid_N_lat) then
+           err_msg = "encountered tile near the poles"
+           call ldas_abort(LDAS_GENERIC_ERROR, Iam, err_msg)
+        end if
+
+        xcur =  real(i_ind(k)-1)*era5_grid_dlon - 180.0 + era5_grid_dlon*0.5
+        ycur =  real(j_ind(k)-1)*era5_grid_dlat -  90.0 + era5_grid_dlat*0.5
+
+        i1(k) = i_ind(k)
+        j1(k) = j_ind(k)
+        x1(k) = xcur    ! lon of grid cell center
+        y1(k) = ycur    ! lat of grid cell center
+
+        ! find forcing grid cell ("2") diagonally across from i_ind(k),j_ind(k)
+
+        tmp_lon = this_lon + 0.5*era5_grid_dlon
+        tmp_lat = this_lat + 0.5*era5_grid_dlat
+        inew =  ceiling((tmp_lon - era5_grid_ll_lon)/era5_grid_dlon)
+        jnew =  ceiling((tmp_lat - era5_grid_ll_lat)/era5_grid_dlat)
+        if (inew==i_ind(k)) inew = inew - 1
+        if (jnew==j_ind(k)) jnew = jnew - 1
+
+        xnew = real(inew-1)*era5_grid_dlon - 180.0+era5_grid_dlon*0.5
+        ynew = real(jnew-1)*era5_grid_dlat -  90.0+era5_grid_dlat*0.5
+        ! wrap-around
+        if (inew==0)              inew = era5_grid_N_lon
+        if (inew>era5_grid_N_lon) inew = 1
+        if ((jnew==0) .or. (jnew>era5_grid_N_lat)) then
+           err_msg = "encountered tile near the poles"
+           call ldas_abort(LDAS_GENERIC_ERROR, Iam, err_msg)
+        end if
+
+        i2(k) = inew
+        j2(k) = jnew
+        x2(k) = xnew    ! lon of grid cell center
+        y2(k) = ynew    ! lat of grid cell center
+
+
+      endif ! if bilinear interpolation
```

Comment:
The above section  duplicates code that is in subroutine GEOS_openfile().
GEOS_openfile() does two things: (i) open GEOS files and (ii) deal with nearest-neighbor and bilinear interpolation from the forcing grid to tile space.  
GEOS_openfile() should be split into two subroutines that each address one of these two jobs.  
Note that the spatial interpolation does not depend on the data coming from GEOS and only needs to know the  grid parameters of the forcing data.  The new spatial interpolation subroutine() can then be used here for the ERA5_LIS forcing to avoid the duplication of code.
cc: @weiyuan-jiang, @biljanaorescanin